stacks-signer/src/coordinator.rs

```diff
@@ -91,17 +91,15 @@ impl CoordinatorSelector {
                 }
             }
             new_index
-        } else {
-            if ROTATE_COORDINATORS {
-                let mut new_index = self.coordinator_index.saturating_add(1);
-                if new_index == self.coordinator_ids.len() {
-                    // We have exhausted all potential coordinators. Go back to the start
-                    new_index = 0;
-                }
-                new_index
-            } else {
-                self.coordinator_index
+        } else if ROTATE_COORDINATORS {
+            let mut new_index = self.coordinator_index.saturating_add(1);
```

Comment:
Not required, but you could simplify this to

```rust
let new_index = self.coordinator_index.saturating_add(1) % self.coordinator_ids.len();
```

and remove the `if` and `mut`.